### PR TITLE
[G2M] Styled Card Patterns

### PR DESCRIPTION
**Pull request by DoParkEQ**

### CODE REVIEW

File: src/styled-checkbox.js [hunk 1]
```diff
@@ -95,7 +95,7 @@ StyledCheckbox.propTypes = {
 StyledCheckbox.defaultProps = {
   checked: false,
   onChange: null,
-  disabled: true,
+  disabled: false,
```
Comment (woozyking) [inline src/styled-checkbox.js:98]: all 3 of these props can be relayed directly from `...rest` as they're all native MUI `Checkbox` props.

File: stories/styled-card-container.stories.js [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+  },
+  sub: {
+    color: theme.palette.grey[600],
+  },
+}));
+
+export const Default = () => {
+  const classes = useStyles();
+
+  const contents = ({ title, main, sub }) => {
```
Comment (woozyking) [inline stories/styled-card-container.stories.js:37]: we can disable props validation for stories altogether, so safe to add that disable clause for this file

File: stories/data/cardInfo.js [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+import img4 from "../../src/assets/fans.png";
+import img5 from "../../src/assets/rogers.jpg";
+import img6 from "../../src/assets/telus.jpg";
+import img7 from "../../src/assets/census.jpeg";
+import img8 from "../../src/assets/custom.jpg";
+import img9 from "../../src/assets/finance.jpg";
+import img10 from "../../src/assets/household.jpg";
+import img11 from "../../src/assets/persona.jpg";
+import img12 from "../../src/assets/retail.jpg";
+import img13 from "../../src/assets/telco.jpg";
```
Comment (woozyking) [inline stories/data/cardInfo.js:13]: are these assets only used for stories? if so, we should move them into `stories/assets` instead to not inflate the library package size upon bundling/publishing

File: src/styled-checkbox.js (hunk 1 above)
Comment (DoParkEQ) [reply src/styled-checkbox.js:98]: Does that mean that those can be removed?

Comment (woozyking) [reply src/styled-checkbox.js:98]: yea both `propTypes` and `defaultProps` can be removed.

you could keep the `defaultProps` if you wish to alter their default behaviour if they're different from MUI

File: src/styled-card-container.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      "&:hover": {
+        transition: "all .6s",
+        boxShadow: theme.palette.shadow[30],
+        backgroundSize: "105% auto",
+      },
+    },
+  };
+});
+
+const StyledCardContainer = ({ pattern, clickable, onClick, children }) => {
```
Comment (hyx131) [inline src/styled-card-container.js:62]: think just `onClick` here would be sufficient

Comment (DoParkEQ) [reply src/styled-card-container.js:62]: What about the default prop settings for onClick? Without clickable, it will always fire the default alert msg.

Comment (hyx131) [reply src/styled-card-container.js:62]: empty function would be okay, like this `() => {}`, correct me if this isn't good practice :)

Comment (DoParkEQ) [reply src/styled-card-container.js:62]: maybe @woozyking can give us some insights on this.

Comment (woozyking) [reply src/styled-card-container.js:62]: depending on the use cases, having `defaultProps` for function `propTypes` can either be no-op function like @hyx131 said, or `null`.

if you need to invoke the function explicitly, a no-op function is needed:
```jsx
<Component
  // this `onClick` needs to be a function, so defaultProps should be () => {}
  onClick={(event) => { onClick(event) }}
/>
```

if you feed that into a native/react virtual event handler, default `null` is safe:
```jsx
<Component
  // onClick event handler accepts null safely
  onClick={onClick}
/>
```

File: stories/data/cardInfo.js [hunk 5]
```diff
@@ -0,0 +1,131 @@
+import img1 from "../assets/branding.jpg";
```
Comment (woozyking) [inline stories/data/cardInfo.js:1]: please keep the files all lower case and use `-` to join words, as suggested https://github.com/EQWorks/common/blob/master/naming.md#code-and-files-associative-naming (reasons within too)

Comment (DoParkEQ) [reply stories/data/cardInfo.js:1]: such as `img-1`?

Comment (woozyking) [reply stories/data/cardInfo.js:1]: sorry i meant this file `cardInfo.js` should be `card-info.js`

File: src/styled-card-container.js (hunk 4 above)
Comment (DoParkEQ) [reply src/styled-card-container.js:62]: So, is it correct that propTypes would be `number` and default props would be `null`?

Comment (woozyking) [reply src/styled-card-container.js:62]: > So, is it correct that propTypes would be `number` and default props would be `null`?

also depends on the use cases, suppose prop name `value` has a type of `number` that's not required.

then in this case when you use it

```jsx
const Component = ({ value }) => {
  const renderFancyValue = () => {
    // if the prop is intended for arithmetics, then it's better to give it a numeral default value (or make it required)
    const calculated = 128 / value

    return (<FancyValue>{caluclated}</FancyValue>)
  }
  return (
    <Paper>
      // if value is used for truthy determination, then defaultProps value to null is better
      {value && renderFancyValue()}
    </Paper>
  )
}
```

File: stories/styled-card-container.stories.js [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+            </CardContent>
+          </StyledCardContainer>
+        </Grid>
+      ))}
+    </Grid>
+  );
+};
+
+const useStyles2 = makeStyles((theme) => {
+  console.log(theme);
```
Comment (YizhiCatherineZhang) [inline stories/styled-card-container.stories.js:68]: Please remove the log if it's not necessary.

**Pull request merged**